functions/src/modules/SE.functions.ts

```diff
@@ -91,71 +91,68 @@ export const processNewAllotment = functions.database
 //
 /////////////////////////////////////////////////
 
-export const uploadProcessing = functions.storage.object()
+export const uploadProcessing = functions.storage.bucket(functions.config().sound_editing.restoration.uploads.bucket_name).object()
 .onFinalize(async object => {    
 
     const filePath = object.name;
-
-
-    if (!filePath.startsWith('sound-editing/restored/uploads'))
-        return -1;
+    const seUploadsBucket = admin.storage().bucket(functions.config().sound_editing.restoration.uploads.bucket_name);    
     
-    
-    const filePathRegex = /sound-editing\/restored\/uploads\/(\w+)\/(\w+)\/(\w+\/)*(([\w\d]+)-\d+-?\d+?)\.flac/;
+    const filePathRegex = /(\w+)\/(\w+\/)*(([\w\d]+)-\d+-\d+)\.flac/;
```

Comment:
Use [non-capturing group](https://javascript.info/regexp-groups#non-capturing-groups-with) to skip the folders in between. This is better than addressing the matches from the end.